server/src/routers/MenuItemsRouter/parseMenuItemData.js

```diff
@@ -0,0 +1,53 @@
+import parseNutritionInfo from './parseNutritionInfo';
+import filters from '../../config/filters.js';
+
+/**
+ * Parses and combines the nutrition (by call to parseNutritionInfo), 
+ * filter, and ingredient data into a single json menu
+ *
+ * throws an error if any of the inputs are null or empty 
+ * or missing DATA field or empty DATA field
+ *
+ * @param {dict} nutritionData
+ * @param {dict} filterData
+ * @param {dict} ingredientData
+ *
+ * return {dict} with menu as a single json object
+ *
+ */
+
+export default function parseMenuItemData(nutritionData, filterData, ingredientData) {
+	// Error checking for invalid/empty data object
+	if (!nutritionData || !nutritionData.DATA || !nutritionData.DATA.length || !nutritionData.DATA[0].length) {
+    	throw new Error('Empty nutrition object returned from YaleDining API');
+    } 
+    if(!filterData || !filterData.DATA || !filterData.DATA.length || !filterData.DATA[0].length){
+    	throw new Error('Empty filter object returned from YaleDining API');
+    } 
+    if(!ingredientData || !ingredientData.DATA || !ingredientData.DATA.length || !ingredientData.DATA[0].length){
+    	throw new Error('Empty ingredient object returned from YaleDining API');
+    }
+
+    // Process ingredient list
+	var ingredientList = [];
+	for (var i = 0; i < ingredientData.DATA.length; i++){
```

Comment:
Nit -- you could do this with a .map() or a .forEach() loop instead, which would match the style of the rest of the backend (and the frontend).

Reply:
Sounds good - changed to .map()